### PR TITLE
i18n translations

### PR DESCRIPTION
**Pull request by gmarcos87**
- [x] Extract rest of English strings into i18n format()
- [ ] Autodetect language
- [ ] Dates and numbers

#485

### CODE REVIEW

File: modules/feed/html/rollup.js [hunk 1]
```diff
@@ -291,44 +293,46 @@ exports.create = function (api) {
...
-      return fn(featuredIds[0])
+  function twoDaysAgo () {
+    return Date.now() - (2 * 24 * 60 * 60 * 1000)
+  }
+
+  function many (ids, fn) {
+    ids = Array.from(ids)
+    var featuredIds = ids.slice(-4).reverse()
+
+    if (ids.length) {
```
Comment (ahdinosaur) [inline modules/feed/html/rollup.js]: hey @gmarcos87, did you know [`intl-messageformat`](https://www.npmjs.com/package/intl-messageformat) which i setup has syntax for plural messages? https://formatjs.io/guides/message-syntax/#plural-format

**Comment by ahdinosaur** [comment]
> [ ] Autodetect language

i [tried to set this up](https://github.com/ssbc/patchwork/pull/485/commits/e9cd1866e71358295790fabd3ad823b05e1f59a5), but didn't really test it. so, does it not work for you?

**Comment by ahdinosaur** [comment]
by the way, this is awesome!

File: locales/en.js [hunk 2]
```diff
@@ -1,6 +1,94 @@
...
+   aProfileLink : "a ProfileLink",
+   chooseProfileImage : "🖼 Choose Profile Image...",
+   chooseName : "Choose a name",
+   describeYourself : "Describe yourself (if you want)",
+   errorPublishing : "An error occured while attempting to publish about message.",
+   inReplyTo : "in reply to",
+   on : "on",
+   referencedFrom : "Referenced from",
+   newMessage : "New Message",
+   likedBy : "Liked by\n",
```
Comment (ralphtheninja) [inline locales/en.js]: From a developer perspective it would be nice if these were "sorted". Just my 2 cents :)

File: locales/en.js [hunk 3]
```diff
@@ -1,6 +1,94 @@
...
+   identified : "identified",
+   identifies : "identifies",
+   as : "as",
+   selfAssignedImage : "self assigned a display image",
+   assignedImageTo : "assigned a display image to",
+   selfAssignedDescription : "self assigned a description",
+   assignedDescriptionTo : "assigned a description to",
+   subscribedTo : "subscribed to",
+   unsubscribedFrom : "unsubscribed from"
+};
```
Comment (ralphtheninja) [inline locales/en.js]: From a developer perspective it would be nice if these key/values were "sorted". Just my 2 cents :)

File: locales/en.js [hunk 4]
```diff
@@ -1,6 +1,94 @@
...
+   publishing : "Publishing...",
+   Publish : "Publish",   
+   Compose : "Compose",
+   clickUnsubscribe : "Click to unsubscribe",
+   subscribe : "Subscribe",
+   subscribed : "Subscribed",
+   writeMessageChannel : "Write a message in this channel\n\n\n\nPeople who follow you or subscribe to this channel will also see this message in their main feed.\n\nTo create a new channel, type the channel name (preceded by a #) into the search box above. e.g #cat-pics",
+   writePrivate : "Write a private reply",
+   writePublic : "Write a public reply",
+   cannotLoadThead : "Cannot load thead",
```
Comment (ralphtheninja) [inline locales/en.js]: Maybe `cannotLoadThread: "Cannot load thread"`?

File: modules/page/html/render/message.js [hunk 5]
```diff
@@ -33,11 +35,11 @@ exports.create = function (api) {
     var compose = api.message.html.compose({
       meta,
       shrink: false,
-      placeholder: when(meta.recps, 'Write a private reply', 'Write a public reply')
+      placeholder: when(meta.recps, format('writePrivate'), format('writePublic'))
     })
 
     api.sbot.async.get(id, (err, value) => {
-      if (err) return result.set(h('div', {className: 'Error'}, ['Cannot load thead']))
+      if (err) return result.set(h('div', {className: 'Error'}, [format('cannotLoadThead')]))
```
Comment (ralphtheninja) [inline modules/page/html/render/message.js]: `cannotLoadThread`

File: modules/feed/html/rollup.js (hunk 1 above)
Comment (ahdinosaur) [inline modules/feed/html/rollup.js]: you should be able to pass in a second argument to `format` which is an object of values you can reference in the translation templates.

File: locales/en.js (hunk 3 above)
Comment (gmarcos87) [inline locales/en.js]: They are in the order they appear in the code, to make the extraction easier. In the next pull I order them. Thank you!

**Comment by ahdinosaur** [comment]
hey @gmarcos87, this is looking great. what's the status on this pull request? is there anything your blocked on, or something you want help with?

**Comment by gmarcos87** [comment]
I have a busy week of work and not be able to make changes in the code. I'm sorry, maybe the next week would work on it.

**Comment by ahdinosaur** [comment]
@gmarcos87 no worries, take care of yourself. :smile_cat:

**Comment by gmarcos87** [comment]
Ok, I'm back. I updated pull request #527 to the current master code. You can see de code in https://github.com/gmarcos87/patchwork/tree/i18n-reload. I extract the strings of the different screens and add the Spanish language. The automatic detection of language and the "moment" configuration are pending.
In the previous pull request (#497) I was using depject and format. Here I continued @dangerousbeans work done on i18n. What approach do you think is best?

I'm really interested in getting a multi-language version. I want to propose the use in my community and not all speak English.

**Comment by dominictarr** [comment]
we should really get one of these merged, I think it's better to use depject, but it should be simple to convert @dangerousbeans's approach to depject

**Comment by gmarcos87** [comment]
Close in favor of #632

**Pull request closed without merging**